src/sentry/runner/commands/cleanup.py

```diff
@@ -310,6 +319,42 @@ def is_filtered(model):
 
                 task_queue.join()
 
+        project_deletion_query = models.Project.objects.filter(status=ObjectStatus.VISIBLE)
+        if project:
+            project_deletion_query = project_deletion_query.filter(project=project)
+
+        for project_id_for_deletion in RangeQuerySetWrapper(
+            project_deletion_query.values_list("id", flat=True),
+            result_value_getter=lambda item: item,
+        ):
+            for model, dtfield, order_by in DELETES_BY_PROJECT:
+                if not silent:
+                    click.echo(
+                        "Removing {model} for days={days} project={project}".format(
+                            model=model.__name__, days=days, project=project_id_for_deletion
+                        )
+                    )
+
+                if is_filtered(model):
+                    if not silent:
+                        click.echo(">> Skipping %s" % model.__name__)
+                else:
+                    imp = ".".join((model.__module__, model.__name__))
+
+                    q = BulkDeleteQuery(
+                        model=model,
+                        dtfield=dtfield,
+                        days=days,
+                        project_id=project_id_for_deletion,
+                        order_by=order_by,
+                    )
+                    q.using = db_router.db_for_read(model, replica=True)
+
+                    for chunk in q.iterator(chunk_size=100):
+                        task_queue.put((imp, chunk))
```

Comment:
The queue has a limit of 1000 entries, but will block with no timeout when full, so this should work fine.